ci/build_wheel.sh

```diff
@@ -33,9 +33,49 @@ rapids-generate-version > ./VERSION
 
 RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
 
+touch /tmp/constraints-build.txt
+
+if [[ "${package_name}" == "cuspatial" ]]; then
+  # Downloads libcuspatial wheel from this current build,
+  # then ensures 'cuspatial' wheel builds always use the 'libcuspatial' just built in the same CI run.
+  RAPIDS_PY_WHEEL_NAME="libcuspatial_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 cpp /tmp/libcuspatial_dist
+  echo "libcuspatial-${RAPIDS_PY_CUDA_SUFFIX} @ file://$(echo /tmp/libcuspatial_dist/libcuspatial_*.whl)" > /tmp/constraints-build.txt
+fi
+
+rapids-logger "Generating build requirements"
+declare -r matrix_selectors="cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION};cuda_suffixed=true"
```

Comment:
Minor nit: Is using `declare` necessary? If we want to start making these kinds of improvements there are probably lots of variables in our CI scripts that should be local and read-only, and perhaps other things too.

Reply:
No not strictly necessary, I was just trying to be explicit about this being read-only.

If you think it's too weird here I'll happily remove it... I definitely don't want to bring "more carefully mark the visibility of all variables in the `ci/` shell scripts" into the scope of this.

Reply:
I don't have a strong preference, it just seemed a bit odd to suddenly introduce this when we aren't careful about this in any of our other scripts. I'm fine keeping it. The only real downside is that it may not be familiar to all readers. I have no interest in increasing the scope of this PR in that direction either. The other change request is much more pertinent for getting this merged.

Reply:
I removed the `declare -r` in https://github.com/rapidsai/cuspatial/pull/1473/commits/29b005edb4d96a826ce8e6b03cabcb6ab5998124, in the interest of consistency with the other scripts and out of a desire to not take on updating all of the references like this.